src/config/field-overrides/program-indicator/AnalyticsPeriodBoundaries.js

```diff
@@ -51,29 +56,52 @@ const boundaryTypes = [
     },
 ];
 
-// boundary type definitions mapped to java enum
-// dhis2-core/dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/AnalyticsPeriodBoundaryType.java#L37-L40
-// as they are not exposed over the API
 function AnalyticsPeriodBoundary(props) {
     const getTranslation = props.d2.i18n.getTranslation.bind(props.d2.i18n);
+
+    const filteredBoundaryTargets = boundaryTargets
+        .filter(e => e.value !== CUSTOM)
+        .filter(e => e.value === props.boundaryTarget)
+        .length
+
+    const renderCustom = () => props.boundaryTarget !== null && filteredBoundaryTargets === 0;
```

Comment:
This covers the "no value" (`null`) option, and if the filtered list is empty, cases which we use to know if we need to render the free-text field.